vpc/vpc-nat-gateway.yaml

```diff
@@ -95,6 +95,74 @@ Resources:
       Dimensions:
       - Name: NatGatewayId
         Value: !Ref NatGateway
+  AlarmNatGatewayBahndwith:
+    Condition: HasAlertTopic
+    Type: 'AWS::CloudWatch::Alarm'
+    Properties:
+      AlarmActions:
+      - {'Fn::ImportValue': !Sub '${ParentAlertStack}-TopicARN'}
+      AlarmDescription: !Sub 'NAT gateway ${SubnetZone} bandwith utilization is over 80%'
+      ComparisonOperator: GreaterThanThreshold
+      EvaluationPeriods: 1
+      Metrics:
+      - Id: 'in1'
+        Label: 'InFromDestination'
+        MetricStat:
+          Metric:
+            Namespace: 'AWS/NATGateway'
+            MetricName: BytesInFromDestination # bytes per minute
+            Dimensions:
+            - Name: NatGatewayId
+              Value: !Ref NatGateway
+          Period: 600
+          Stat: Average
+          Unit: Bytes
+        ReturnData: false
+      - Id: 'in2'
+        Label: 'InFromSource'
+        MetricStat:
+          Metric:
+            Namespace: 'AWS/NATGateway'
+            MetricName: BytesInFromSource # bytes per minute
+            Dimensions:
+            - Name: NatGatewayId
+              Value: !Ref NatGateway
+          Period: 600
+          Stat: Average
+          Unit: Bytes
+        ReturnData: false
+      - Id: 'out1'
+        Label: 'OutToDestination'
+        MetricStat:
+          Metric:
+            Namespace: 'AWS/NATGateway'
+            MetricName: BytesOutToDestination # bytes per minute
+            Dimensions:
+            - Name: NatGatewayId
+              Value: !Ref NatGateway
+          Period: 600
+          Stat: Average
+          Unit: Bytes
+        ReturnData: false
+      - Id: 'out2'
+        Label: 'OutToSource'
+        MetricStat:
+          Metric:
+            Namespace: 'AWS/NATGateway'
+            MetricName: BytesOutToSource # bytes per minute
+            Dimensions:
+            - Name: NatGatewayId
+              Value: !Ref NatGateway
+          Period: 600
+          Stat: Average
+          Unit: Bytes
+        ReturnData: false
+      - Expression: '(in1+in2+out1+out2)/60*8/1000/1000/1000' # to Gbit/s
```

Comment:
The expression counts traffic twice, doesn't it?

Reply:
I wasn't sure about this as well. https://aws.amazon.com/premiumsupport/knowledge-center/cloudwatch-nat-gateway-bandwidth/ states that we have to add all 4 numbers.

I performed a test:
EC2 a in private subnet
EC2 b in public subnet

I was sending 0.5 Gibt/s from a to b and measured the exact same number with the formula. So I guess it's correct?

Reply:
Let me create a support request.

Reply:
ok. I have to divide by 2 :)